src/backend/CassandraBackendNew.h

```diff
@@ -0,0 +1,943 @@
+ * other parts such as ETL do not have to change at all.
+ * Eventually we should change the interface so that it does not have to know
+ * about yield_context.
+ */
+template <
+    SomeSettingsProvider SettingsProviderType,
+    SomeExecutionStrategy ExecutionStrategy>
+class BasicCassandraBackend : public BackendInterface
+{
+    clio::Logger log_{"Backend"};
+
+    SettingsProviderType settingsProvider_;
+    Schema<SettingsProviderType> schema_;
+    Handle handle_;
+
+    // have to be mutable because BackendInterface constness :(
+    mutable ExecutionStrategy executor_;
+
+    std::uint32_t ledgerSequence_ = 0;
+
+public:
+    /**
+     * @brief Create a new cassandra/scylla backend instance.
+     *
+     * @param settingsProvider
+     */
+    BasicCassandraBackend(SettingsProviderType settingsProvider)
+        : settingsProvider_{std::move(settingsProvider)}
+        , schema_{settingsProvider_}
+        , handle_{settingsProvider_.getSettings()}
+        , executor_{settingsProvider_.getSettings(), handle_}
+    {
+        if (auto const res = handle_.connect(); not res)
+            throw std::runtime_error(
+                "Could not connect to Cassandra: " + res.error());
+        if (auto const res = handle_.execute(schema_.createKeyspace); not res)
+            throw std::runtime_error(
+                "Could not create keyspace: " + res.error());
+        if (auto const res = handle_.executeEach(schema_.createSchema); not res)
+            throw std::runtime_error("Could not create schema: " + res.error());
+
+        schema_.prepareStatements(handle_);
+        log_.info() << "Created (revamped) CassandraBackend";
+    }
+
+    /*! Not used in this implementation */
+    void
+    open([[maybe_unused]] bool readOnly) override
+    {
+    }
+
+    /*! Not used in this implementation */
+    void
+    close() override
+    {
+    }
+
+    TransactionsAndCursor
+    fetchAccountTransactions(
+        ripple::AccountID const& account,
+        std::uint32_t const limit,
+        bool forward,
+        std::optional<TransactionsCursor> const& cursorIn,
+        boost::asio::yield_context& yield) const override
+    {
+        auto rng = fetchLedgerRange();
+        if (!rng)
+            return {{}, {}};
+
+        Statement statement = [this, forward, &account]() {
+            if (forward)
+                return schema_->selectAccountTxForward.bind(account);
+            else
+                return schema_->selectAccountTx.bind(account);
+        }();
+
+        auto cursor = cursorIn;
+        if (cursor)
+        {
+            statement.bindAt(1, cursor->asTuple());
+            log_.debug() << "account = " << ripple::strHex(account)
+                         << " tuple = " << cursor->ledgerSequence
+                         << cursor->transactionIndex;
+        }
+        else
+        {
+            auto const seq = forward ? rng->minSequence : rng->maxSequence;
```

Comment:
one thing not related to this PR: seq is not really used.

Reply:
it's only used for the log line in this same block of code.. it costs nothing to have it in a constant tho vs. inlining the ternary. i'll leave it as is for now.